packages/devtools_app/lib/src/table.dart

```diff
@@ -117,18 +119,36 @@ class FlatTableState<T> extends State<FlatTable<T>>
     sortData(widget.sortColumn, widget.sortDirection);
   }
 
-  List<double> _computeColumnWidths(double maxWidth) {
+  @visibleForTesting
+  List<double> computeColumnWidths(double maxWidth) {
+    // Subtract width from outer padding around table.
     maxWidth -= 2 * defaultSpacing;
+    maxWidth -= widget.numSpacers * defaultSpacing;
+    maxWidth = max(0, maxWidth);
 
     final unconstrainedCount =
         widget.columns.where((col) => col.fixedWidthPx == null).length;
-    final allocatedWidth = widget.columns.fold(0,
-        (val, col) => col.fixedWidthPx == null ? val : col.fixedWidthPx + val);
-    final available = maxWidth - allocatedWidth;
+    double allocatedWidth = 0;
+    for (var col in widget.columns) {
+      if (col.fixedWidthPx != null) {
+        allocatedWidth += col.fixedWidthPx;
+      } else {
+        if (col.minWidthPx != null) {
+          allocatedWidth += col.minWidthPx;
+        }
+      }
+    }
+    final available = max(0, maxWidth - allocatedWidth);
 
     final widths = <double>[];
     for (ColumnData<T> column in widget.columns) {
-      final width = column.fixedWidthPx ?? (available / unconstrainedCount);
+      double width = column.fixedWidthPx;
+      if (width == null) {
+        width = available / unconstrainedCount;
+        if (column.minWidthPx != null) {
+          width += column.minWidthPx;
+        }
+      }
       widths.add(width);
```

Comment:
at the end of this for loop, should `width` be subtracted from `available`? Otherwise the available space could be inaccurate for the remaining unconstrained columns

Reply:
no because the number of unconstrained columns is not changed. Changing the # of unconstrained columns would be even stranger as the the relative sizes of columns would depend on the order of the columns.